pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -177,9 +177,22 @@ func (m *migration) getNotificationChannelMap() (channelsPerOrg, defaultChannels
 
 // Create a notifier (PostableGrafanaReceiver) from a legacy notification channel
 func (m *migration) createNotifier(c *notificationChannel) (*PostableGrafanaReceiver, error) {
-	uid, err := m.generateChannelUID()
-	if err != nil {
-		return nil, err
+	uid := c.Uid
+	if uid == "" {
```

Comment:
sorry if this is a dumb question, but under which circumstances would this happen? We always generate new contact points based on existing ones IIRC, so there's no need for the code below.

Reply:
This idea of checking to make sure the uid isn't empty and potentially using the ID as a backup has been around since the [original migration code](https://github.com/grafana/grafana/commit/e52f718a87f9f26813bfe45cd30d823e7a1d42f7#diff-3d6f33c060b6bf9fcfcb6c64e52c81198f5793d5ee389dd81395633cd8ea50f8R54-R59) and as far as I know the reason has been lost to to time.

I kept it around "just in case" when I refactored the migration code a while back. IMO it makes some sense to also make sure we're not copying an empty uid here if we still have this [code elsewhere](https://github.com/grafana/grafana/blob/0db339d82f905a7f4c0db11717bcb22b7fd9f111/pkg/services/sqlstore/migrations/ualert/channel.go#L231-L237).

Reply:
Thanks, @JacobsonMT! Indeed, it was mostly just to guard against weird edge cases. I actually didn't know we had similar checks elsewhere, that's a great point. Sure enough, it seems like empty string UIDs are a very valid concern.

I also have no idea how strict Legacy alerting's object validation is. A lot of resources in Grafana let you control the UID, and we've seen more than one case where empty string was allowed, usually unintentionally. Maybe a handful of users out there have an empty string as a UID already? Now, we'll ensure that all the UIDs make sense.

Reply:
Makes sense, thanks for the answers!